### PR TITLE
Add application commands badge

### PR DESCRIPTION
**Pull request by HyperGH**
### Summary

Add the new application commands badge to `ApplicationFlags`.

### Checklist
- [x] I have run `nox` and all the pipelines have passed.
- [ ] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/applications.py [hunk 1]
```diff
@@ -95,6 +95,9 @@ class ApplicationFlags(enums.Flag):
     MESSAGE_CONTENT_INTENT_LIMITED = 1 << 19
     """Denotes that the application has message content access while pending verification."""
 
+    APPLICATION_COMMAND_BADGE = 1 << 23
+    """Denotes that the application supports app commands and has the corresponding badge."""
```
Comment (davfsa) [inline hikari/applications.py:99]:
```suggestion
    """Denotes that the application has at least one application command."""
```

**Review by davfsa — CHANGES_REQUESTED** [review]
#

File: hikari/applications.py (hunk 1 above)
Comment (HyperGH) [reply hikari/applications.py:99]: If we want to be exact, then it should be "has at least one **global** application command".

File: hikari/applications.py [hunk 2]
```diff
@@ -95,6 +95,9 @@ class ApplicationFlags(enums.Flag):
     MESSAGE_CONTENT_INTENT_LIMITED = 1 << 19
     """Denotes that the application has message content access while pending verification."""
 
+    APPLICATION_COMMAND_BADGE = 1 << 23
+    """Denotes that the application has at least one global application command, and thus has the corresponding badge."""
```
Comment (davfsa) [inline hikari/applications.py:99]:
```suggestion
    """Denotes that the application has at least one global application command."""
```
 The rest of the information is redundant. It is already implied in the context

**Review by davfsa — CHANGES_REQUESTED** [review]

**Review by davfsa — APPROVED** [review]

**Comment by FasterSpeeding** [comment]
Btw might've jumped the gun on merging this just a tad bit considering this is an undocumented feature and that announcement said they were "testing" this feature

**Comment by davfsa** [comment]
Oh shoot. I am currently not home and I saw the announcement and just assumed they had rolled it out. My bad

**Comment by HyperGH** [comment]
The announcement mentions that it was activated yesterday for everyone, so I assumed it was okay to make a PR for it.

**Pull request merged**